scrapy/utils/reqser.py

```diff
@@ -90,10 +92,12 @@ def _find_method(obj, func):
             pass
         else:
             if func_self is obj:
-                name = func.__func__.__name__
-                if _is_private_method(name):
-                    return _mangle_private_name(obj, func, name)
-                return name
+                members = inspect.getmembers(obj, predicate=inspect.ismethod)
+                for name, obj_func in members:
+                    if obj_func.__func__ is func.__func__:
```

Comment:
Could you please explain (add a comment) why isn't it `if obj_func is func`?

Reply:
```
We need to use __func__ to access the original
function object because instance method objects
are generated each time attribute is retrieved from
instance.

Reference: The standard type hierarchy 
https://docs.python.org/3/reference/datamodel.html
```